lib/pi_piper/sysfs/driver.rb

```diff
@@ -0,0 +1,106 @@
+require 'filewatcher'
+
+module PiPiper
+  module Sysfs
+    class Driver < PiPiper::Driver
+
+      GPIO_HIGH = 1
+      GPIO_LOW  = 0
+
+      def initialize
+        @exported_pins = Set.new
+      end
+
+      def close
+        unexport_all
+        @exported_pins.empty? 
+      end
+      
+  # Support GPIO pins
+      def pin_direction(pin, direction)
+        raise ArgumentError, "direction should be :in or :out" unless [:in, :out].include? direction
+        export(pin)
+        raise RuntimeError, "Pin #{pin} not exported" unless exported?(pin)
+        File.write(direction_file(pin), direction)
+      end
+
+      def pin_read(pin)
+        raise ArgumentError, "Pin #{pin} not exported" unless exported?(pin)
+        File.read(value_file(pin)).to_i
+      end
+
+      def pin_write(pin, value)
+        raise ArgumentError, "value should be GPIO_HIGH or GPIO_LOW" unless [GPIO_LOW, GPIO_HIGH].include? value
+        raise ArgumentError, "Pin #{pin} not exported" unless exported?(pin)
+        File.write(value_file(pin), value)
+      end
+
+      def pin_set_pud(pin, value)
+        raise NotImplementedError, "Pull up/down not available with this driver. keep it on :off" unless value == :off
+      end
+      
+      def pin_set_trigger(pin, trigger)
+        raise ArgumentError, "trigger should be :falling, :rising, :both or :none" unless [:falling, :rising, :both, :none].include? trigger
+        raise ArgumentError, "Pin #{pin} not exported" unless exported?(pin)
+        File.write(edge_file(pin), trigger)
+      end
+
+      def pin_wait_for(pin, trigger)
+        pin_set_trigger(pin, trigger)
+        value = pin_read(pin)
+
```

Comment:
Here are some stackoverflow questions I was researching when testing the original implementation:

http://stackoverflow.com/questions/11367719/why-is-my-ruby-script-utilizing-90-of-my-cpu
http://linux.die.net/man/2/select
(there are a couple more, I will have to re-search them on google but will link them once I get home)

Apparently `select` does not block because the `read` does not block when at `EOF` but rather returns `nil`. This is why it consumes the CPU. If you look at the man page for select it even states that it will only block if the read blocks, this is not the case for us. Read will always succeed.